docs/pages/access-controls/sso/login-rules.mdx

```diff
@@ -0,0 +1,863 @@
+---
+title: Transform user traits with Login Rules
+description: Set up Login Rules to transform traits from your IDP
+---
+
+This guide explains how to filter, transform, and extend incoming SAML
+assertions and OIDC claims from your identity provider during user login.
+
+<VarList />
+
+## Introduction
+
+Teleport populates each SSO user's **traits** from their SAML assertions,
+OIDC claims, or parts of their GitHub identity during login.
+These traits can be referenced in
+[Role Templates](../guides/role-templates/)
+in order to provide differing levels of access to users. 
+
+You can see your current user's traits by running the following command while
+logged in to your cluster:
+```code
+$ tctl get --format json users/<Var name="username"/> | jq 'first.spec.traits'
+{
+  "groups": [
+    "devs",
+    "dbs"
+  ],
+  "username": [
+    "Alice"
+  ]
+}
+```
+
+**Login rules** can be used to transform the traits provided by your IDP to meet your
+needs for configuring access within Teleport.
+
+Some usecases for login rules are:
+
+- Users in group `A` should also be added to group `B`. Login Rules can
+  easily add extra trait values based on logical rules, without needing to make
+  any changes in your IDP.
+- Your IDP provides too many traits and user certificates are too large. Login
+  Rules can filter down your user traits to only those which are necessary.
+- Login rules can define a single trait based on a logical combination of other
+  traits, which can easily be used in multiple role templates without
+  repetition.
+
+This guide will walk through an example scenario where the following changes
+need to be made to incoming traits:
+
+1. Every user with the `groups: devs` trait should receive an extra trait
+   `access: [staging]`.
+1. Every user with the `groups: admins` trait should receive an extra trait
+   `access: [staging, prod]`.
+1. Every user should receive a `logins` trait with the value of their incoming
+   `username` trait converted to lowercase.
+
+## Prerequisites
+
+Before you get started you’ll need a Teleport v11.3+ cluster with an existing
+OIDC, SAML, or GitHub connector configured and working for user logins.
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/5. Find all traits currently in use in your cluster
+
+Before creating a login rule, find out all of the traits currently in use by
+templated roles in your cluster.
+You can use the following command to find all such traits.
+If using trusted clusters, run this command for all connected clusters to make
+sure all traits are found.
+
+```code
+$ tctl get roles | grep 'internal\|external'
+```
+
+When creating any login rule, make sure that all of these traits are mapped so
+that they are not inadvertently removed.
+
+## Step 2/5. Draft your login rule resource.
+
+Copy the following text to `my_rule.yaml`, making any changes necessary for your
+cluster.
+
+[Read on](#expression-syntax) for details on the expression syntax.
```

Comment:
If we're expecting the reader to follow each step sequentially, I don't think we should require the reader to skip around the guide as well. Can we hint that we'll explain the expression syntax below without linking the reader to another part of the guide?

As a side note, I think the Expression Syntax section works better as a separate guide. If we go with this approach, I think we should add the link to the Expression Syntax guide to a "Next Steps" section at the end of the how-to guide.

Reply:
Split out the syntax into a reference page and simplified the guide a bit